app/app.js

```diff
@@ -20,12 +23,51 @@ app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, '../public')));
+//app.use(express.bodyParser()); need?
 
 app.get('/', function(req, res, next) {
     console.log('routing to index');
     res.render('index');
     });
 
+app.post('/api/v1/beers', function(req, res, next) {
+    // Construct the brewerydb endpoint to hit
+    endpoint = 'http://api.brewerydb.com/v2/search?q=' + encodeURIComponent(req.body.beer) + '&key=' + secrets.bdb_key
+    console.log(endpoint);
+
+    if (req.body.mock) {
+        res.send(JSON.stringify(mocks.one_beer));
+    }
+    else {
+        // Hit the endpoint
+        request(endpoint, function (error, response, body) {
+            if (!error && response.statusCode == 200) {
+                var data = JSON.parse(body).data;
+                var matches = 0
+
+                // Create an array that filters the return data such that the last
+                // entry in the array will be the data entry that had the most words
+                // that matched the original queried beer
+                var qualifiers = data.filter( function(el) {
```

Comment:
This current filtering mechanism has potential to be wasteful, let's clean it up with a `break` algorithm instead.